lib/BackgroundJobs/FileSystemListenerJob.php

```diff
@@ -44,45 +49,60 @@ protected function run($argument): void {
 			$this->diagnosticService->sendHeartbeat(static::class, $this->getId());
-
 			try {
-				$node = current($this->rootFolder->getUserFolder($fsEvent->getUserId())->getById($fsEvent->getNodeId()));
-			} catch (\Exception $e) {
-				$this->logger->warning('Error retrieving node for fs event "' . $fsEvent->getType() . '": ' . $e->getMessage(), ['exception' => $e]);
-				$node = false;
+				$fsEvents = $this->fsEventMapper->getFromQueue(static::BATCH_SIZE);
+			} catch (Exception $e) {
+				$this->logger->warning('Error fetching fs events: ' . $e->getMessage(), ['exception' => $e]);
+				return;
 			}
-			if ($node === false) {
-				$this->logger->warning('Node with ID ' . $fsEvent->getNodeId() . ' not found for fs event "' . $fsEvent->getType() . '"');
-				$this->fsEventMapper->delete($fsEvent);
-				continue;
+
+			if (empty($fsEvents)) {
+				return;
 			}
 
-			try {
-				switch ($fsEvent->getTypeObject()) {
-					case FsEventType::CREATE:
-						$this->fsEventService->onInsert($node);
-						break;
-					case FsEventType::ACCESS_UPDATE_DECL:
-						$this->fsEventService->onAccessUpdateDecl($node);
-						break;
-				}
+			foreach ($fsEvents as $fsEvent) {
 				$this->diagnosticService->sendHeartbeat(static::class, $this->getId());
-				$this->fsEventMapper->delete($fsEvent);
-			} catch (\RuntimeException $e) {
-				$this->logger->warning('Error handling fs event "' . $fsEvent->getType() . '": ' . $e->getMessage(), ['exception' => $e]);
+
+				try {
+					$node = current($this->rootFolder->getUserFolder($fsEvent->getUserId())->getById($fsEvent->getNodeId()));
+				} catch (\Exception $e) {
+					$this->logger->warning('Error retrieving node for fs event "' . $fsEvent->getType() . '": ' . $e->getMessage(), ['exception' => $e]);
+					$node = false;
+				}
+				if ($node === false) {
+					$this->logger->warning('Node with ID ' . $fsEvent->getNodeId() . ' not found for fs event "' . $fsEvent->getType() . '"');
+					try {
+						$this->fsEventMapper->delete($fsEvent);
+					} catch (Exception $e) {
+						$this->logger->warning('Error deleting fs event "' . $fsEvent->getType() . '": ' . $e->getMessage(), ['exception' => $e]);
+					}
+					continue;
+				}
+
+				try {
+					try {
+						switch ($fsEvent->getTypeObject()) {
+							case FsEventType::CREATE:
+								$this->fsEventService->onInsert($node);
+								break;
+							case FsEventType::ACCESS_UPDATE_DECL:
+								$this->fsEventService->onAccessUpdateDecl($node);
+								break;
+						}
+						$this->diagnosticService->sendHeartbeat(static::class, $this->getId());
+					} catch (\RuntimeException $e) {
+						$this->logger->warning('Error handling fs event "' . $fsEvent->getType() . '": ' . $e->getMessage(), ['exception' => $e]);
+					}
+					$this->fsEventMapper->delete($fsEvent);
+				} catch (Exception $e) {
+					$this->logger->warning('Error deleting fs event "' . $fsEvent->getType() . '": ' . $e->getMessage(), ['exception' => $e]);
+				}
```

Comment:
the try block for this can start from line 99
and the catch block of the inner try..catch could catch all the exceptions `\Exception`.
that could potentially make it a little easier to read, wdyt?